aixplain/processes/data_onboarding/process_media_files.py

```diff
@@ -0,0 +1,223 @@
+from aixplain.utils.file_utils import upload_data
+from pathlib import Path
+from tqdm import tqdm
+from typing import List, Tuple
+
+
+def compress_folder(folder_path: str):
+    with tarfile.open(folder_path + ".tgz", "w:gz") as tar:
+        for name in os.listdir(folder_path):
+            tar.add(os.path.join(folder_path, name))
+    return folder_path + ".tgz"
+
+
+def run(metadata: MetaData, paths: List, folder: Path, batch_size: int = 100) -> Tuple[List[File], int, int, int, int]:
+    """Process a list of local media files, compress and upload them to pre-signed URLs in S3
+
+    Explanation:
+        Each media on "paths" is processed. If the media is in a public link, this link is added into an index CSV file.
+        If the media is in a local path, it will be copied into a local folder and its path will be added to the index CSV file.
+        The medias are processed in batches such that at each "batch_size" medias, the index CSV file is uploaded into a pre-signed URL in s3 and reset.
+        If the medias are stored locally, the local folder is compressed into a .tgz file and also uploaded into S3.
+
+    Args:
+        metadata (MetaData): meta data of the asset
+        paths (List): list of paths to local files
+        folder (Path): local folder to save compressed files before upload them to s3.
+
+    Returns:
+        Tuple[List[File], int, int, int]: list of s3 links; data, start and end columns index, and number of rows
+    """
+    # if files are stored locally, create a folder to store it
+    media_folder = Path(".")
+    if metadata.storage_type == StorageType.FILE:
+        media_folder = Path(os.path.join(folder, "data"))
+        media_folder.mkdir(exist_ok=True)
+
+    idx = 0
+    data_column_idx, start_column_idx, end_column_idx = -1, -1, -1
+    files, batch, start_intervals, end_intervals = [], [], [], []
+    for i in tqdm(range(len(paths)), desc=f' Data "{metadata.name}" onboarding progress', position=1, leave=False):
+        path = paths[i]
+        try:
+            dataframe = pd.read_csv(path)
+        except Exception as e:
+            message = f'Data Asset Onboarding Error: Local file "{path}" not found.'
+            logging.exception(message)
+            raise Exception(message)
+
+        # process medias
+        for j in tqdm(range(len(dataframe)), desc=" File onboarding progress", position=2, leave=False):
+            row = dataframe.iloc[j]
+            try:
+                media_path = row[metadata.name]
+            except Exception as e:
+                message = f'Data Asset Onboarding Error: Column "{metadata.name}" not found in the local file "{path}".'
+                logging.exception(message)
+                raise Exception(message)
+
+            # adding medias
+            if metadata.storage_type == StorageType.FILE:
+                if metadata.dtype == DataType.AUDIO:
+                    assert (
+                        os.path.getsize(media_path) <= 50000000
```

Comment:
Declare all constants right after the imports in the top of the file and name them.